docs/NEW_NEAR_CLI_INTERFACE.md

```diff
@@ -0,0 +1,191 @@
+# New NEAR CLI interface
+
+`NEAR CLI` is using `extensions` to saticfy all groups of users.
+- `Core NEAR CLI` commands should be usefull for all groups of users.
+- `Extensions` are used by a particular group or several groups of users.
+- `Core NEAR CLI` is a single binary, there is no extensions that are installed `by default`.
+- Each extension is a separate binary that can be installed and executed from `NEAR CLI`.
+- `Core NEAR CLI` extensions system only allows to introduce top-level commands through its design.
+
+`NEAR CLI` is built for:
+- NEAR `dApp developers`, who build smart-contracts, UIs, and tooling on NEAR.
+- `Tech-savvy` people automating their routines.
+- `Validators`
+
+`NEAR CLI UX principles`
+- All altering actions should have a confirmation step with an option to skip confirmation with an explicit command line parameter (e.g. `send` at the end of the command)
+- All direct children commands of a single parent command should be aligned (either represent an action or a resource, but never a mix of those on the same hierarchy level): `contract` -> `state` (resource) and `contract` -> `deploy` (action) are not aligned, so it should be either `contract` -> `get-state` + `contract` -> `deploy` or `contract` -> `state` -> `view` + `contract` -> `code` -> `deploy`
+- Interactive mode should look like: `command - description`. It will help people to learn the commands.
+
+## `Core NEAR CLI` commands
+
+```
+account
+  - view-account <account-id> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+ 
+  - create-sub-account <new-account-id> <initial-balance> 
+    (we will treat everything after the first dot in account id as the parent account (transaction signer))
+    - with-plaintext-public-key "ed25519:..." network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+    - with-generated-keypair network <"mainnet"|"testnet"|...>
```

Comment:
Same as above but key pair will be generated during command execution? It would be cool to merge them, but optional arguments are not supported by the current design.

Reply:
> Same as above but key pair will be generated during command execution?

Yep, that is right.

> It would be cool to merge them, but optional arguments are not supported by the current design.

I don't think we need to merge them. These are different options just like options of how to sign the transaction.